pkg/controller/pxc/users.go

```diff
@@ -326,3 +327,41 @@ func sysUsersSecretDataChanged(newHash string, usersSecret *corev1.Secret) (bool
 func sha256Hash(data []byte) string {
 	return fmt.Sprintf("%x", sha256.Sum256(data))
 }
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileSyncPXCUsersWithProxySQL(cr *api.PerconaXtraDBCluster) {
+	if cr.Status.Status != api.AppStateReady {
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	stopCh := make(chan error, 1)
+
+	go func() {
+		r.syncUsersChan <- 1
```

Comment:
can me moved outside of goroutine to be sure that we always have value in `syncUsersChan` and no more need to do check `if len(r.syncUsersChan) > 0`

Reply:
we can't be sure that `syncUsersChan` will have value, because we run this function  in separate Goroutine and context can finish it earlier

Reply:
>can be moved outside of goroutine to be sure that we always have value in syncUsersChan